Scripts/build.swift

```diff
@@ -117,6 +117,7 @@ for rawPlatform in rawPlatforms {
         "-configuration", "Release",
         "-derivedDataPath", platform.derivedDataPath,
         "-PBXBuildsContinueAfterErrors=0",
+        "OTHER_CFLAGS='-DGENERATED_XCODE_PROJECT'",
```

Comment:
What does this do?

Reply:
And why is it necessary? (Asking to learn)

Reply:
This enables us to check `#if GENERATED_XCODE_PROJECT` in code. It enables the following code:
https://github.com/dfed/CacheAdvance/blob/345ca001533be5599aa2e000ea4a9134fbb0b1da/Tests/CADCacheAdvanceTests/CADCacheAdvanceTests.m#L18-L26

We need this because the generated Xcode project requires different import styles than required by compiling when opening up the  `Package.swift`. This is a rough edge of Swift Package Manager that I ran into in this PR.

Reply:
Good to know.
Asking to learn: why do we generate the Xcode project and build that way, vs. building using SPM?
Do we have any CI tests that verify the SPM builds directly (i.e. that don't generate the Xcode project)? I didn't seen any when I briefly looked.

Reply:
I don't know how to use the command line to run-and-build on specific simulators using SPM directly, so I generate the project and use `xcodebuild` (which I'm pretty familiar with). It's possible we could simplify a good bit by relying on `swift` rather than `xcodebuild`, but I don't know and the ROI of finding out seems small?

I do not have any tests that directly test SPM. That could make a good addition.

Reply:
https://github.com/dfed/CacheAdvance/issues/44